android/app/src/main/java/app/organicmaps/widget/placepage/DirectionFragment.java

```diff
@@ -45,11 +46,15 @@ protected int getCustomTheme()
   public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState)
   {
     final View root = inflater.inflate(R.layout.fragment_direction, container, false);
+    final MwmActivity mwm_activity = (MwmActivity) requireActivity();
     root.setOnTouchListener((v, event) -> {
+      root.performClick();
+      mwm_activity.setUiHidden(false);
```

Comment:
Where is this DirectionFragment displayed/destroyed? Would it be more logical to show/hide the UI there?

Reply:
It's created in placepageview.java, and destroyed by itself in it's onclicklistener.

I guess it could be moved to placepageview, but we'll always want to try to hide the placepage/buttons because the fragment has been designed to overlay the map.
IMO it's easier to ensure that behaviour by calling the hide from the view onCreate, so if we call it from somewhere else in the future it'll hide correctly.

Then again, I'm very far from an expert 😅 if you think it's better practice to do it in placepageview i'll do it :)

Reply:
Maybe @organicmaps/android have some ideas? I'm a bit confused by using mwm activity as the global variable here, but maybe it's a right approach.